content/en/blog/2019/isolated-clusters/index.md

```diff
@@ -0,0 +1,1880 @@
+    gateways:
+      enabled: true
+      istio-egressgateway:
+        enabled: false
+      istio-ingressgateway:
+        enabled: false
+      istio-private-egressgateway:
+        enabled: true
+        labels:
+          app: istio-private-egressgateway
+          istio: private-egressgateway
+        replicaCount: 1
+        autoscaleMin: 1
+        autoscaleMax: 5
+        cpu:
+          targetAverageUtilization: 80
+        type: ClusterIP
+        ports:
+        - port: 80
+          name: http
+        - port: 443
+          name: https
+        secretVolumes:
+        - name: c1-example-com-certs
+          secretName: c1-example-com-certs
+          mountPath: /etc/istio/c1.example.com/certs
+        - name: c1-trusted-certs
+          secretName: c1-trusted-certs
+          mountPath: /etc/istio/example.com/certs
+    EOF
+    {{< /text >}}
+
+1.  Verify that the egress gateway's pod is running:
+
+    {{< text bash >}}
+    $ kubectl get pods $(kubectl get pod -l istio=private-egressgateway -n istio-private-gateways -o jsonpath='{.items..metadata.name}' --context=$CTX_CLUSTER1)  -n istio-private-gateways --context=$CTX_CLUSTER1
+    NAME                                           READY   STATUS    RESTARTS   AGE
+    istio-private-egressgateway-586c8cb5db-5m77h   1/1     Running   0          43s
+    {{< /text >}}
+
+1.  Verify that the key and the certificates are successfully loaded in the egress gateway's pod:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -l istio=private-egressgateway -n istio-private-gateways -o jsonpath='{.items..metadata.name}' --context=$CTX_CLUSTER1)  -n istio-private-gateways --context=$CTX_CLUSTER1 -- ls -al /etc/istio/c1.example.com/certs /etc/istio/example.com/certs
+    /etc/istio/c1.example.com/certs:
+    total 4
+    drwxrwxrwt 3 root root  120 Jul 29 00:27 .
+    drwxr-xr-x 3 root root 4096 Jul 29 00:27 ..
+    drwxr-xr-x 2 root root   80 Jul 29 00:27 ..2019_07_29_00_27_05.153388454
+    lrwxrwxrwx 1 root root   31 Jul 29 00:27 ..data -> ..2019_07_29_00_27_05.153388454
+    lrwxrwxrwx 1 root root   14 Jul 29 00:27 tls.crt -> ..data/tls.crt
+    lrwxrwxrwx 1 root root   14 Jul 29 00:27 tls.key -> ..data/tls.key
+
+    /etc/istio/example.com/certs:
+    total 4
+    drwxrwxrwt 3 root root  100 Jul 29 00:27 .
+    drwxr-xr-x 3 root root 4096 Jul 29 00:27 ..
+    drwxr-xr-x 2 root root   60 Jul 29 00:27 ..2019_07_29_00_27_05.678454477
+    lrwxrwxrwx 1 root root   31 Jul 29 00:27 ..data -> ..2019_07_29_00_27_05.678454477
+    lrwxrwxrwx 1 root root   22 Jul 29 00:27 example.com.crt -> ..data/example.com.crt
+    {{< /text >}}
+
+### Deploy a private ingress gateway in the second cluster
+
+1.  Create the `istio-private-gateways` namespace:
+
+    {{< text bash >}}
+    $ kubectl apply --context=$CTX_CLUSTER2 -f - <<EOF
```

Comment:
Why not the one liner `kubectl --context=$CTX_CLUSTER2 create namespace istio-private-gateways` whenever you need to create a namespace?

Reply:
@esnible This is done to prevent an error if the command to create the namespace is run twice in the case of a single cluster (once for CTX_CLUSTER1 and once for CTX_CLUSTER2, which is the same cluster in case of the single cluster).

BTW, running the instructions on a single cluster was your idea and I like it very much. All the instructions can be checked and tested on a single cluster, and it is actually an example of isolation of namespaces within the same cluster.